rc/core/formatter.kak

```diff
@@ -1,13 +1,19 @@
 decl -docstring "shell command to which the contents of the current buffer is piped" \
     str formatcmd
 
-def format -docstring "Format the contents of the current buffer" %{ eval -draft %{
-    %sh{
-        if [ -n "${kak_opt_formatcmd}" ]; then
-            ## Save the current position of the cursor
-            readonly x=$((kak_cursor_column - 1))
-            readonly y="${kak_cursor_line}"
+# This option holds the selection as it was at the time of calling the `format` command
+# After the buffer has been formatted, the selection will be restored according to
+# the value held in this option
+decl -hidden range-specs formatter_prevsel
```

Comment:
I dont think that will work until piping does diff the input and the output and uses that as the actual modifications. Currently piping just replaces the input with the output, which means update-option will not give use anything interesting.